### PR TITLE
Fix for being unable to charge UPS and use UPS-modded tools without UPS

### LINKED ISSUE #14976: UPS-compatible recharging station doesn't charge UPS
For Build 4237:
The UPS-compatible recharging station doesn't charge UPSs or advanced UPSs.

The station however **does** charge items modded with the "UPS conversion mod" **but** those refuse to use that charge.
For example: Equip the UPS conversion mod to a jackhammer, charge it with the station and use it. It will refuse to work without a charged UPS in your inventory. (Despite being charged itself).

### PR DESCRIPTION
**Pull request by Core0verload**
Battery-powered UPS is now rechargeable by default. It can be charged in any recharging station, but you can't unload it anymore. The same applies for two items that are supposed to have rechargeable power cells: e-tonfa (it's stated in description) and control laptop (regular laptop has rechargeable cells in it's disassembly). However, you can't charge a plutonium-powered UPS that way.

All UPS-modded tools now could be used without UPS, making UPS mod a complete replacement for rechargeable battery mod, as it was intended.

UPS charging station is added to Humvee's box. Other mil. vehicles already have it.

Fixes #14976.

### CODE REVIEW

**Comment by Core0verload** [comment]
>Somebody PR something that gives us a rechargeable UPS option - could be a new UPS-specific mod or a new variant of the UPS item itself - and I'll mergetest it.

@Rivet-the-Zombie, here we go.

**Comment by Rivet-the-Zombie** [comment]
Thank you, @Core0verload! I'll get on this ASAP.

**Comment by Rivet-the-Zombie** [comment]
The game gives no prior warning that you can't remove batteries once you load them into a UPS, so that's going to be a surprise for unwary people. Please include something in the item's description to let them know.

Other than that, everything is fine.

**Comment by Malkeus** [comment]
:/ I still can't see the benefit of the change. Originally ups could be
loaded and unloaded. Adding the rechargeable mod to a ups made it
un-unloadable (probably un-loadable too, I never tried). Which was fine, it
worked with minimum headaches. Change for the sake of change isn't great
imo, n I would have spoken up against this one if it had been discussed.
Discussion would have at least given me some idea of the perceived benefit
of the change.

**Comment by Rivet-the-Zombie** [comment]
A UPS-specific recharging mod would be a good fix for this whole situation; the vanilla UPS wouldn't have to change, and it would give us more options for UPS power.

**Comment by Malkeus** [comment]
This ^  . I've been struggling to articulate exactly what I objected to
about the changes and that covers it nicely. We lost control of the ups
with this change. The changes to the recharging mod are fine, I can live
with them, but there wasn't anything wrong with the ups functionality
before this change.

**Comment by kevingranade** [comment]
I might not be following, but is the functionality you want back that the
UPS can create batteries on demand as long as it's charged? If so that is
precisely the thing we wanted to get rid of.  The UPS isn't a box you put
batteries in, it's a battery pack itself, so you should be charging and
discharging it, not adding and removing batteries.

**Comment by Rivet-the-Zombie** [comment]
I think the gist is that folks want three varieties of UPS:

One that you load/unload with batteries.
One that you load/unload with plutonium cells.
One that you charge with a charging station and can't load/unload otherwise.

We already have the first two types; if we add a UPS-specific mod that works on the first type, it would turn the first type into the third type.

Did that cover the basics, @Malkeus?

**Comment by Malkeus** [comment]
That is not what I'm asking for at all. The original way it worked was a
ups without a recharge mod allowed batteries to be loaded and unloaded.
With the recharge mod you could neither load or unload, it only gained
charge in the recharge station.
That option to operate on either batteries or car power is what was lost.
If there was an exploit there I didn't know about it. Which circles right
back around to things that would have been mentioned during a discussion
which never took place.

**Comment by Core0verload** [comment]
It's not like unloading UPS is used very often. And everybody was using them with rechargeable mod anyway, so it makes sense to make it rechargeable by default. I don't understand that "we lost control" thing at all.

**Comment by Malkeus** [comment]
@Rivet yep, that covers it. I often find newer characters have more
electronics than batteries to run then with. A ups that's a battery black
hole would require a lot of inventory management to avoid loading it with
more than you can afford to spare. A reload dialogue that let you specify
how many batteries to load would mitigate this somewhat, instead of forcing
you to drop all of them but the ones you wanted to load. Not ideal though.

**Comment by Malkeus** [comment]
Who is this everyone you speak of? I agree that unloading is not an issue
in mid to late game, but it comes up a lot in the first few days of my
playthroughs.

**Comment by Core0verload** [comment]
UPS spawns empty, you can't just loot it for power cells in early game. And using UPS as a power source in the first days isn't very common because you had to find an UPS, some UPS-powered tools and some batteries, with only batteries being common enough for an early survivor to get.

**Comment by Malkeus** [comment]
Rngesus smiles on us from time to time. Not common doesn't feel like a
valid reason to render something impossible. Kevin cites the fact that it's
a battery pack and thus should be incapable of accepting batteries. I
disagree, there should be a middle ground which works with batteries. I'm
on my phone so I can't look up the description, is Kevin's assertion
accurate as far as the description goes? I seem to recall something about
batteries.

**Comment by Core0verload** [comment]
>This is a unified power supply, or UPS.  It is a device developed jointly by military and scientific interests for use in combat and the field.  The UPS is designed to power armor and some guns, but drains batteries quickly.

This is UPS's desc.

**Comment by Malkeus** [comment]
That description is vague enough to support any interpretation. Kevin said
rechargeable battery packs; I've always viewed them as reloadable power box
attached to a wireless power transmitter. The  wireless power transmission
device was the specially developed part. I dunno. I digress.

All I'm really asking for here is some supporting logic for the change. And
your unsourced opinion that everyone is using it a certain way is not that.

**Comment by Core0verload** [comment]
And I thought the discussion about storage batteries balance was pointless...

The only reason behind these changes: people were complaining a lot about UPS being non-rechargeable. I didn't even knew it was possible to make it rechargeable and that a lot of people actually used that. So I made it rechargeable. And what's the supporting logic behind your nitpicking?

**Comment by Malkeus** [comment]
That the whole change was unnecessary precisely because it was already
possible to make them rechargeable. Which is something you just pointed out
that you didn't even know about. It worked before, it worked well and after
you changed it, it was broken. Now your 'fix' leaves it less capable than
it was before.

I like the enhancement to the recharging station. I like combining ups and
recharge mods for tools. Those are both good ideas and enhance the game.
The sole thing I disagree with is the alterations to the ups BC it is now
less capable than its previous incarnation.

On a personal level, the fact that you call discussion nitpicking, get
angry over said discussion AND couldn't be bothered to thoroughly research
and then test your own PR bothers me.

You changed something to make it capable of doing something it was already
capable of doing. And not for the better, imo.

**Comment by Core0verload** [comment]
So you think that unloading is a major feature of an UPS, and removing it made UPS a lot less capable than it was before, even if it was replaced with an ability to recharge in station. I don't think so. The further discussion will be pointless.

**Comment by Malkeus** [comment]
I think you 'fixed' something that wasn't broken. You can be as dismissive as you like, but it doesn't change the FACT that you didn't know the feature you added already existed. And for the record, I never said it was a **major** feature, nor that it was a** lot** less capable, don't put words into my mouth. You still haven't given a reason for the change, btw.

**Comment by Malkeus** [comment]
http://smf.cataclysmdda.com/index.php?topic=12053.msg266096#msg266096
This is some more of that pointless discussion.

**Comment by Core0verload** [comment]
If you don't like it, feel free to open a pull that removes ["RECHARGE", "NO_UNLOAD"] from a regular UPS and adds a recipe for converting the regular UPS into a rechargeable.

**Comment by Malkeus** [comment]
So mote it be.

File: data/json/items/tools.json [hunk 1]
```diff
@@ -4236,7 +4238,7 @@
     "color": "light_green",
     "name": "advanced UPS",
     "name_plural": "advanced UPS's",
-    "description": "This is an advanced version of the unified power supply, or UPS.  This device has been significantly redesigned to provide better efficiency as well as to consume plutonium fuel cells rather than batteries.",
+    "description": "This is an advanced version of the unified power supply, or UPS.  This device has been significantly redesigned to provide better efficiency as well as to consume plutonium fuel cells rather than batteries.  Sadly, it's plutonium reactor can't be charged in UPS charging station.",
```
Comment (Coolthulhu) [inline data/json/items/tools.json]: Should be "its", not "it's".

**Pull request merged**